File: 11-container-with-most-water/11-container-with-most-water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        left = 0
        right = len(height) - 1
        max_area = 0
        while (left < right):
            temp = (right - left) * min(height[left] , height[right])
            max_area = max(temp ,max_area)
            if height[left] < height[right]:
                left += 1
            else:
                right -= 1            
        return max_area

File: 11-container-with-most-water/11-container-with-most-water.java
class Solution {
    public int maxArea(int[] height) {
        int left = 0;
        int right = height.length - 1;
        int max_area = 0;
        while (left < right){
            int temp = (right - left) * Math.min(height[left] , height[right]);
            max_area = Math.max(temp ,max_area);
            if (height[left] < height[right]){
                left += 1; }
            else{
                right -= 1; }
        }            
        return max_area;
    }
}